pages/modPanel.php

```diff
@@ -671,6 +677,162 @@ function modPanel() {
         $logs = $player->staff_manager->getStaffLogs('staff_logs', StaffManager::STAFF_LOG_MOD, $offset, $limit);
         require_once 'templates/staff/mod/mod_logs.php';
     }
+    //Mod stats
+    else if($view == 'mod_stats' && $player->staff_manager->isHeadModerator()) {
+        $display_menu = false;
+        $self_link = $system->router->getUrl('mod', ['view'=>'mod_stats']);
+        $PREVIOUS_MONTH = $system->SERVER_TIME;
+        $PREVIOUS_MONTH = $PREVIOUS_MONTH->modify('-30 days');
+        $min_year = (int) $system->SERVER_TIME->modify('-5years')->format('Y');
+        $max_year = (int) $system->SERVER_TIME->format('Y');
+        $months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+
+        if(isset($_POST['load_stats'])) {
+            try {
+                $mod_staff = [];
+                $moderator_staff_levels = [StaffManager::STAFF_MODERATOR, StaffManager::STAFF_HEAD_MODERATOR, StaffManager::STAFF_ADMINISTRATOR, StaffManager::STAFF_HEAD_ADMINISTRATOR];
+                $mod_result = $system->db->query("SELECT `user_name`, `user_id` FROM `users` WHERE `staff_level` IN (" . implode(', ', $moderator_staff_levels) . ")");
+                if (!$system->db->last_num_rows) {
+                    echo "No staff m8.";
+                }
+
+                // Validate day to month
+                $start_day = $_POST['start_day'];
+                $end_day = $_POST['end_day'];
+
+                if($_POST['start_month'] == 'Feb') {
+                    $max_days = 28;
+                    if((($_POST['start_year'] % 4) == 0) && ((($_POST['start_year'] % 100) != 0) || (($_POST['start_year'] %400) == 0))) {
+                        $max_days = 29;
+                    }
+                    $start_day = min($start_day, $max_days);
+                }
+                if($_POST['end_month'] == 'Feb') {
+                    $max_days = 28;
+                    if((($_POST['end_year'] % 4) == 0) && ((($_POST['end_year'] % 100) != 0) || (($_POST['end_year'] %400) == 0))) {
+                        $max_days = 29;
+                    }
+                    $start_day = min($start_day, $max_days);
+                }
+                if(in_array($_POST['start_month'], ['Apr', 'Jun', 'Sep', 'Nov'])) {
+                    $start_day = min($start_day, 30);
+                }
+                if(in_array($_POST['end_month'], ['Apr', 'Jun', 'Sep', 'Nov'])) {
+                    $end_day = min($end_day, 30);
+                }
+
+                // Start from beginning of day these must be reported in GMT as logs are processed as such
+                $start_string = $_POST['start_month'] . ' ' . $start_day . ' ' . $_POST['start_year'] . ' 00:00:00';
```

Comment:
Assuming someone used SQL injection on this field so the value of `$_POST['start_month']` is like `'; DELETE FROM users;--"`, I wouldn't expect DateTimeImmutable to pass through the text, but I don't know the internals, so let's sanitize to be safe: 
```suggestion
                $start_string = (int)$_POST['start_month'] . ' ' . $start_day . ' ' . (int)$_POST['start_year'] . ' 00:00:00';
```

Reply:
In my testing, malformed date-time strings failed to initialize.
I think we could move that down a bit further where the time stamps are pulled?

```php
                // Set stamps for queries
                $start_time_stamp = (int) $start_time->getTimestamp();
                $end_time_stamp = (int) $end_time->getTimestamp();
```